src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -162,7 +162,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
             if (ParameterHelpers.RequiresLifetimeAnnotationAttribute(this))
             {
-                AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeLifetimeAnnotationAttribute(this, Scope));
+                AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeLifetimeAnnotationAttribute(this, DeclaredScope));
```

Comment:
>DeclaredScope [](http://example.com/codeflow?start=114&length=13)

nit: there's not really a difference but this feels more like EffectiveScope

Reply:
I agree, there shouldn't be a difference here, but I think I prefer `DeclaredScope` since that is what was declared.

Reply:
Still nit but want to clarify my reasoning: Declared scope is only a step towards producing the real/effective scope. I think most places should use the real/effective scope unless there's a special reason. When it comes to emitting (like here), clearly we should use the real/effective scope, not its precedessor. If we added any more factors into the computations of real/effective scope, we'd want them to affect what we emit.

Reply:
Currently, we're emitting implicitly scoped parameters with no `[ScopedRef]` attribute. If we use `EffectiveScope` for emit, then emitting `[UnscopedRef] out int i` (where the `EffectiveScope` is `Unscoped`) will require ensuring an `[UnscopedRef]` attribute is emitted, perhaps by assuming an `[UnscopedRef]` attribute was specified in source. Instead, if we use `DeclaredScope` here (which is perhaps poorly named since `DeclaredScope` is (implicitly )`RefScoped` for `out int i`), then any `[UnscopedRef]` from source will provided the expected result in metadata.

A better alternative might be to emit implicitly scoped parameters _with_ a `[ScopedRef]` attribute. Then `EffectiveScope` may be simpler here.